Import v8 wasm mjsunit tests

msaboff opened the pull request:
#### f25b969b98a842b3cd1ee6a084ffb59229e1e6b1
<pre>
Import v8 wasm mjsunit tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=249562">https://bugs.webkit.org/show_bug.cgi?id=249562</a>
rdar://103498866

Reviewed by Nobody.

Import and add the v8 mjsunit Wasm tests.
This includes adding some new resource .js files and adding those resource files to run-jsc-stress-tests.
Created a script, Tools/Scripts/convert-imported-v8-tests.py, to automate much of the process.
This script searches for calls to v8 internal &apos;%&apos; functions, calls to print, load directives and
either converts them to JSC equivalents or flags the tests with &quot;skip&quot; directives.
Of the 200+ files, over 165 have been skipped for various issues.  In all of these skipped tests,
comments haave been added to the top of the test file with the reason they were skipped.

Canonical link: <a href="https://commits.webkit.org/258088@main">https://commits.webkit.org/258088@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e916cdf47991e321579fb63f1718d63bee22b13e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100873 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/10025 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33921 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110179 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10956 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/888 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93287 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/108022 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106656 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/8284 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/91542 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/34900 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/90192 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22938 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/77875 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/91359 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3714 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/24459 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/87447 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/1242 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3731 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/851 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/29238 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9844 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/43954 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/90335 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5509 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/20223 "Passed tests") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2907 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian reviewed (approved):
rs=me.  Let's also get @Constellation's opinion on which directory to put these imported tests given he had already previously imported some.

Constellation reviewed (approved):
The current path looks good to me.

msaboff committed:
Import v8 wasm mjsunit tests
https://bugs.webkit.org/show_bug.cgi?id=249562
rdar://103498866

Reviewed by Nobody.

Import and add the v8 mjsunit Wasm tests.
This includes adding some new resource .js files and adding those resource files to run-jsc-stress-tests.
Created a script, Tools/Scripts/convert-imported-v8-tests.py, to automate much of the process.
This script searches for calls to v8 internal '%' functions, calls to print, load directives and
either converts them to JSC equivalents or flags the tests with "skip" directives.
Of the 200+ files, over 165 have been skipped for various issues.  In all of these skipped tests,
comments haave been added to the top of the test file with the reason they were skipped.

Canonical link: https://commits.webkit.org/258088@main
